R/functions-Spectrum1.R

```diff
@@ -102,60 +103,63 @@ Spectra1_mz_sorted <- function(peaksCount = NULL, rt = numeric(),
     if (length(mz) != length(intensity))
         stop("Lengths of 'mz' and 'intensity' do not match!")
     nvals <- length(nvalues)
+    empty_int <- rep(NA_integer_, nvals)
+    empty_num <- as.numeric(empty_int)
+    empty_log <- as.logical(empty_int)
     ## Now match all of the lengths to the length of nvalues.
-    if (length(peaksCount) == 0)
+    if (!length(peaksCount))
         peaksCount <- nvalues
     ## rt
-    if (length(rt) == 0){
-        rt <- rep(NA_integer_, nvals)
+    if (!length(rt)) {
+        rt <- empty_int
```

Comment:
Are all this `if(!length(...))` statements really necessary? I have no clue about the input values. Would it be possible to set the default argument of this constructor to e.g. `rt=rep_len(NA_integer_, length(nvalues))`?

Reply:
I know, it's ugly this way. The *problem* is that the upstream function uses e.g. `fdata$polarity` to pass the polarity column from the feature data to the function. If the `featureData` has been reduced and column `"polarity"` was removed, `fdata$polarity` is `NULL` and `NULL` is then passed with the `polarity` parameter.